version: "3.9"

services:
  app:
    build: .
    ports:
      - "4000:4000"
    env_file:
      - .env.dev
    networks:
      - track-search-network
    depends_on:
      - mongodb
    volumes:
      - .:/usr/src/app
      - /usr/src/app/node_modules

  mongodb:
    image: mongo:6.0.3
    container_name: mongodb
    command: ["--replSet", "rs0", "--bind_ip_all", "--port", "27017"]
    environment:
      MONGO_INITDB_DATABASE: track-search-database
    ports:
      - "27017:27017"
    healthcheck:
      test: echo "try { rs.status() } catch (err) { rs.initiate({_id:'rs0',members:[{_id:0,host:'host.docker.internal:27017'}]}) }" | mongosh --port 27017 --quiet
      interval: 5s
      timeout: 30s
      start_period: 0s
      retries: 30
    volumes:
      - mongodb_data:/data/db
    networks:
      - track-search-network

  mongo-gui:
    image: ugleiton/mongo-gui
    container_name: mongo-gui
    restart: always
    environment:
      MONGO_URL: mongodb://mongodb:27017/track-search-database?replicaSet=rs0
    volumes:
      - mongodb_data:/data/db
    networks:
      - track-search-network
    ports:
      - "4321:4321"
    depends_on:
      - mongodb

volumes:
  mongodb_data:


networks:
  track-search-network:
